File: ContractService/Models/DepositDispute.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ContractService.Models
{
    public class DepositDispute
    {
        [Key]
        public int DepositDisputeId { get; set; }
        [ForeignKey("DisputeStatus")]
        public int DisputeStatusId { get; set; }

        public decimal Deposit { get; set; }
        public decimal? DepositWithheld { get; set; }
        public DisputeStatus DisputeStatus { get; set; }
    }
}

File: ContractService/ClientApp/src/app/models/depositDispute.ts
export interface DepositDispute {
    depositDisputeId: number;
    disputeStatusName: string;
    deposit: number;
    depositWithheld: number;
}

export interface DisputeUpdate {
    depositDisputeId: number;
    deposit: number;
    depositWithheld: number;
}

export interface DisputeMessage {
    depositDisputeId: number;
    depositWithheld: number;
    WitheldReason: string;
}

